driver/registry_sql.go

```diff
@@ -86,7 +86,7 @@ func (m *RegistrySQL) Init(ctx context.Context) error {
 		}
 
 		if m.C.HsmEnabled() {
-			hardwareKeyManager := hsm.NewKeyManager(m.HsmContext())
+			hardwareKeyManager := hsm.NewKeyManager(m.HsmContext(), m.C.HsmKeyPrefix())
```

Comment:
Could you instead please pass the function itself, or alternatively `m.C`? That way, hot reloading this config value would work.